Ability to change Search Engine

Infinixius opened the pull request:
This PR adds the ability to change the search engine from Google, to something like DuckDuckGo or YouTube.

Adding new search engines is incredibly easy, just add a new entry to `engines` in `index.js`. For example:
```json
{
  "name": "YouTube",
  "url": "https://youtube.com/",
  "search": "https://youtube.com/results?search_query=",
  "icon": "../assets/img/ytLogo.png"
}
```

![image](https://user-images.githubusercontent.com/68125679/137563204-bf00455b-be14-4cfa-a875-6f6c58ec16c7.png)
![image](https://user-images.githubusercontent.com/68125679/137563207-f8890903-5b0f-419a-912d-75061ec56767.png)
![image](https://user-images.githubusercontent.com/68125679/137563210-24d4ede5-addd-4b28-91f7-870d53bcb6ca.png)

Lateralus138 commented:
I mentioned the need for this at Dev.To when this was first released so I, of course, second this request. *DDG* is a must lol.

virejdasani commented:
@Infinixius, thanks a lot for contributing, you have implemented it beautifully, just a few things:
- You have used `localStorage`, this isn't approved by the chrome webstore or the firefox addon store. They have different APIs:
- - https://developer.chrome.com/docs/extensions/reference/storage/
- - https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/storage/local
- The text in the DDG search buttons is not vertically centered
- The switch search engine button is on the very edge of the display. Could you maybe add it in the left button modal?

Infinixius commented:
> 
> 
> @Infinixius, thanks a lot for contributing, you have implemented it beautifully, just a few things:
> 
>     * You have used `localStorage`, this isn't approved by the chrome webstore or the firefox addon store. They have different APIs:
> 
>     * * https://developer.chrome.com/docs/extensions/reference/storage/
> 
>     * * https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/storage/local
> 
>     * The text in the DDG search buttons is not vertically centered
> 
>     * The switch search engine button is on the very edge of the display. Could you maybe add it in the left button modal?

I'm still not sure if the Chrome/Firefox independent versions I added work as intended. Fixed the DDG's vertical centering issue and moved the switch search engine button into the modal.



> 
> 
> Also, what do you think of the buttons? Would they look better without the hover glow?

I think they look just fine with the glow.

virejdasani commented:
@Infinixius, this looks awesome! Thanks a lot for contributing 👍

virejdasani commented:
@all-contributors please add @Infinixius for code
